### ISSUE TITLE
Get rid of references to old admin tools

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3423]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#3423|
|Type|defect|
|Created|2008-09-08 22:48:15Z|
|Branch|https://github.com/twisted/twisted/tree/process-docs-3423|

```
./admin/update-documentation.py:    dumps it (ie, admin/process-docs has been run) and into a WebSite working
./doc/development/policy/svn-dev.xhtml:<code>admin/process-docs</code> shell script. This will create processed
./doc/development/policy/svn-dev.xhtml:done in <code>admin/process-docs</code>, such as the following. Note that
./doc/lore/howto/lore.xhtml:<code>admin/process-docs</code> script.</span></p>
./emacs/twisted-dev.el:;;   f12: run admin/process-docs
./emacs/twisted-dev.el:    (compile (format "%s/admin/process-docs" twisted-dev-directory))))
./twisted/lore/test/test_lore.py:#   admin/process-docs
```

<details><summary>Searchable metadata</summary>

```
trac-id__3423 3423
type__defect defect
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_process_docs_3423 branches/process-docs-3423
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1220914095000000 1220914095000000
changetime__1221615564000000 1221615564000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [24843]) Branching to 'process-docs-3423'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [thijs](#3423):
> ./admin/update-documentation.py:    dumps it (ie, admin/process-docs has been run) and into a WebSite working

Removed this reference in r24844

> ./doc/development/policy/svn-dev.xhtml:<code>admin/process-docs</code> shell script. This will create processed
> ./doc/development/policy/svn-dev.xhtml:done in <code>admin/process-docs</code>, such as the following. Note that

svn-dev.xhtml mentions 'To generate the full documentation set, run the admin/process-docs shell script. This will create processed HTML, man pages, and 250-page book.pdf' but that script was killed, so whats the alternative?

> ./doc/lore/howto/lore.xhtml:<code>admin/process-docs</code> script.</span></p>

Same here, what can be used to 'generate the HTML version of the Twisted docs from a SVN checkout'?

> ./emacs/twisted-dev.el:;;   f12: run admin/process-docs
> ./emacs/twisted-dev.el:    (compile (format "%s/admin/process-docs" twisted-dev-directory))))

Is this script still being used?

> ./twisted/lore/test/test_lore.py:#   admin/process-docs

Should the note on the top of this file be removed or rewritten?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Replying to [thijs](#note_2):
> svn-dev.xhtml mentions 'To generate the full documentation set, run the admin/process-docs shell script. This will create processed HTML, man pages, and 250-page book.pdf' but that script was killed, so whats the alternative?
>

There is no single replacement for `process-docs`.  That script did several things:

  * convert the xhtml lore input into html output with lore
  * convert the troff man pages into html output with lore
  * Generate the pdf version of the book

`twisted.python._release` now contains APIs for doing all of these things but there are no command line tools which wrap these them.  These APIs mainly just invoke entrypoints of tools which do have command line interfaces, though.  The xhtml and troff files are converted by invoking lore's main point.  The book is a bit more complicated, since several steps are involved, and one of the tools is latex.  It is _not_ intended that users invoke these APIs, so where possible the documentation should not refer to them.  However, for the case of the book, it may make sense (I leave it up to you to judge whether the required chain of lore/latex/dvips/ps2pdf13 commands is too complicated for developers to follow).  Generating the book is also rarely something people are interested in doing, so we may just say "read the code or come back later when we actually have a tool".

> > ./doc/lore/howto/lore.xhtml:<code>admin/process-docs</code> script.</span></p>
>
> Same here, what can be used to 'generate the HTML version of the Twisted docs from a SVN checkout'?

This is just `lore`.  It needs to be run from `doc/core/howto` in order to find the template (or you can pass the template path as a parameter).  It accepts filenames on the command line, or it will walk the current directory recursively finding .xhtml files and processing them.

>
> > ./emacs/twisted-dev.el:;;   f12: run admin/process-docs
> > ./emacs/twisted-dev.el:    (compile (format "%s/admin/process-docs" twisted-dev-directory))))
>
> Is this script still being used?
>

Not by me.  I think it may be save to drop F12 from twisted-dev.

> > ./twisted/lore/test/test_lore.py:#   admin/process-docs
>
> Should the note on the top of this file be removed or rewritten?
>

Removed.  Unit tests is unit tests, there's nothing else. (ug the top of that file is terrible, we have so much work to do).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_3):
> There is no single replacement for `process-docs`.  That script did several things:
>
>   * convert the xhtml lore input into html output with lore
>   * convert the troff man pages into html output with lore
>   * Generate the pdf version of the book
>
> `twisted.python._release` now contains APIs for doing all of these things but there are no command line tools which wrap these them.  These APIs mainly just invoke entrypoints of tools which do have command line interfaces, though.  The xhtml and troff files are converted by invoking lore's main point.  The book is a bit more complicated, since several steps are involved, and one of the tools is latex.  It is _not_ intended that users invoke these APIs, so where possible the documentation should not refer to them.  However, for the case of the book, it may make sense (I leave it up to you to judge whether the required chain of lore/latex/dvips/ps2pdf13 commands is too complicated for developers to follow).  Generating the book is also rarely something people are interested in doing, so we may just say "read the code or come back later when we actually have a tool".

I removed that paragraph for now and I'll open a new ticket for the missing command line tool wrappers, which can bring back that paragraph in the future. See r24865

> This is just `lore`.  It needs to be run from `doc/core/howto` in order to find the template (or you can pass the template path as a parameter).  It accepts filenames on the command line, or it will walk the current directory recursively finding .xhtml files and processing them.

Removed reference in r24866

> Not by me.  I think it may be save to drop F12 from twisted-dev.

Removed it in r24864

> Removed.  Unit tests is unit tests, there's nothing else. (ug the top of that file is terrible, we have so much work to do).

Removed in r24866

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24880]) Merge process-docs-3423: remove references to old admin/process-docs script.

Author: thijs
Reviewer: exarkun
Fixes: [#3423](https://github.com/twisted/twisted/issues/3423)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
